File: backend/app/data/default_skills.py
"""Standard D&D 5e skills seeded for every user."""

from __future__ import annotations

# (name, attribute code)
DEFAULT_SKILLS: tuple[tuple[str, str], ...] = (
    ("Athletics", "STR"),
    ("Acrobatics", "DEX"),
    ("Sleight of Hand", "DEX"),
    ("Stealth", "DEX"),
    ("Arcana", "INT"),
    ("History", "INT"),
    ("Investigation", "INT"),
    ("Nature", "INT"),
    ("Religion", "INT"),
    ("Animal Handling", "WIS"),
    ("Insight", "WIS"),
    ("Medicine", "WIS"),
    ("Perception", "WIS"),
    ("Survival", "WIS"),
    ("Deception", "CHA"),
    ("Intimidation", "CHA"),
    ("Performance", "CHA"),
    ("Persuasion", "CHA"),
)

DEFAULT_SKILL_NAMES: frozenset[str] = frozenset(
    name.casefold() for name, _ in DEFAULT_SKILLS
)


def is_default_skill_name(name: str) -> bool:
    return name.casefold() in DEFAULT_SKILL_NAMES

File: frontend/lib/features/player_options/skills/data/default_skills.dart
/// Keep in sync with [backend/app/data/default_skills.py].
class DefaultSkill {
  const DefaultSkill({required this.name, required this.attribute});

  final String name;
  final String attribute;
}

const defaultSkills = <DefaultSkill>[
  DefaultSkill(name: 'Athletics', attribute: 'STR'),
  DefaultSkill(name: 'Acrobatics', attribute: 'DEX'),
  DefaultSkill(name: 'Sleight of Hand', attribute: 'DEX'),
  DefaultSkill(name: 'Stealth', attribute: 'DEX'),
  DefaultSkill(name: 'Arcana', attribute: 'INT'),
  DefaultSkill(name: 'History', attribute: 'INT'),
  DefaultSkill(name: 'Investigation', attribute: 'INT'),
  DefaultSkill(name: 'Nature', attribute: 'INT'),
  DefaultSkill(name: 'Religion', attribute: 'INT'),
  DefaultSkill(name: 'Animal Handling', attribute: 'WIS'),
  DefaultSkill(name: 'Insight', attribute: 'WIS'),
  DefaultSkill(name: 'Medicine', attribute: 'WIS'),
  DefaultSkill(name: 'Perception', attribute: 'WIS'),
  DefaultSkill(name: 'Survival', attribute: 'WIS'),
  DefaultSkill(name: 'Deception', attribute: 'CHA'),
  DefaultSkill(name: 'Intimidation', attribute: 'CHA'),
  DefaultSkill(name: 'Performance', attribute: 'CHA'),
  DefaultSkill(name: 'Persuasion', attribute: 'CHA'),
];
